index.html

```diff
@@ -0,0 +1,261 @@
+              rows[index].cells[2].childNodes[0].checked = item[2];
+              rows[index].cells[3].childNodes[0].value = item[3] !== '1000-01-01' ? item[3] : '';
+  
+          });
+          sortASC[col] = !sortASC[col];
+      }
+  
+      function addStringFieldEvent(row) {
+          row.cells[0].childNodes[0].addEventListener('input', function (event) {
+              var input = event.target,
+                  value = input.value,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+  
+              tbl[rowNum][0] = value;
+          });
+      }
+      function addNumberFieldEvent(row) {
+          row.cells[1].childNodes[0].addEventListener('keypress', function (event) {
+              var charCode = (event.which) ? event.which : event.keyCode;
+              if (charCode < 48 || charCode > 57) {
+                  event.preventDefault();
+              }
+          });
+          row.cells[1].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target,
+                  value = input.value,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+              if (value === '') {
+                  value = Number.MAX_SAFE_INTEGER;
+              }
+              tbl[rowNum][1] = parseInt(value);
+          });
+      }
+      function addBooleanFieldEvent(row) {
+          row.cells[2].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target,
+                  checked = input.checked,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+  
+              tbl[rowNum][2] = checked;
+          });
+      }
+      function addDateFieldEvent(row) {
+          row.cells[3].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target || event.srcElement,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1,
+                  date = input.value;
+  
+              tbl[rowNum][3] = date;
+          });
+      }
+      function addDeleteFieldEvent(row) {
+          row.cells[4].childNodes[0].addEventListener('click', function (event) {
+              var input = event.target || event.srcElement,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1,
+                  table = document.querySelector("#tblBody");
+  
+              table.deleteRow(rowNum);
+              tbl.splice(rowNum, 1);
+          });
+      }
+  
+      function assignEvents(row) {
+          addStringFieldEvent(row);
+          addNumberFieldEvent(row);
+          addBooleanFieldEvent(row);
+          addDateFieldEvent(row);
+          addDeleteFieldEvent(row);
+      }
+  
+      function addNameCell(nameCell) {
```

Comment:
You are talking about adding the cells?
Its really the same, evert cell has its own properties..